### PR TITLE
Coffee Shop

### PR DESCRIPTION
**Pull request by Hubblenaut**
## Coffee Shop Update
Image
  - Adds coffee dispenser
  - Adds sugar dispensers
  - Adds porcelaine cups
  - Remaps the coffee shop
  - Saltshaker, peppermill and new sugar dispenser are now condiment/small

**Todo:** We totally need vendors with baked goods.

### CODE REVIEW

**Comment by Minijar** [comment]
Muffin shop when?

**Comment by Yoshax** [comment]
More like bakery when?

File: code/modules/reagents/dispenser/cartridge_presets.dm [hunk 1]
```diff
@@ -62,6 +60,14 @@
 	lime		spawn_reagent = "limejuice"
 	watermelon	spawn_reagent = "watermelonjuice"
 
+	// Bar, coffee
+	coffee		spawn_reagent = "coffee"
+	cafe_latte	spawn_reagent = "cafe_latte"
+	soy_latte	spawn_reagent = "soy_latte"
+	hot_coco	spawn_reagent = "hot_coco"
+	milk		spawn_reagent = "milk"
```
Comment (GinjaNinja32) [inline code/modules/reagents/dispenser/cartridge_presets.dm]: New cartridge types should be made orderable in supply.dm

File: code/modules/reagents/dispenser/dispenser_presets.dm [hunk 2]
```diff
@@ -113,3 +113,22 @@
...
+/obj/machinery/chemical_dispenser/bar_coffee/full
+	spawn_cartridges = list(
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/coffee,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/cafe_latte,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/soy_latte,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/hot_coco,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/milk,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/cream,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/tea,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/ice
```
Comment (GinjaNinja32) [inline code/modules/reagents/dispenser/dispenser_presets.dm]: Cartridges will not regenerate over time with this setup; you'll need to add the type in dispenser2_energy.dm for that to happen.

**Comment by Hubblenaut** [comment]
Done.

**Comment by Earthcrusher** [comment]
Oh man, that looks nice. And I'm glad someone got to this because I was slaaaackin'.

File: code/modules/reagents/dispenser/cartridge_presets.dm (hunk 1 above)
Comment (GinjaNinja32) [inline code/modules/reagents/dispenser/cartridge_presets.dm]: Not addressed yet.

**Comment by Draconomial** [comment]
Needs a laptop dispenser outside it.

**Comment by NortonDK** [comment]
Could use a wall clipboard

File: code/modules/reagents/dispenser/cartridge_presets.dm (hunk 1 above)
Comment (Hubblenaut) [inline code/modules/reagents/dispenser/cartridge_presets.dm]: It's in supply.dm

Comment (GinjaNinja32) [inline code/modules/reagents/dispenser/cartridge_presets.dm]: You've only done a crate of all of them; scroll down to PACK amd SEC_PACK and add the  there too, so they can be ordered separately.

Comment (Hubblenaut) [inline code/modules/reagents/dispenser/cartridge_presets.dm]: And done.

**Comment by Zuhayr** [comment]
Conflicts.

**Comment by OhEmGee** [comment]
Wait, this PR doesn't add the "coffee boy" job?

**Comment by Zuhayr** [comment]
Please reopen this when you have updated your branch so that this can be merged.

**Comment by Hubblenaut** [comment]
If nobody mind we could have a coffee boy alt title.

**Comment by Zuhayr** [comment]
An official title for 'coffee boy' is unnecessary imo.

**Comment by Woodratt** [comment]
Barista is already a alt title for the bartender.

**Comment by Hubblenaut** [comment]
Anyhow, this is ready for merging.

**Comment by Neerti** [comment]
What happens if you, say, put lime juice in a coffee cup?  I might be reading this wrong but it looks like weird stuff would happen if ```cup_name``` and such were undefined for the reagent.

**Comment by GinjaNinja32** [comment]
I should revive my drinking glass rewrite and make it properly generic

*nod*

**Comment by Hubblenaut** [comment]
> What happens if you, say, put lime juice in a coffee cup? I might be reading this wrong but it looks like weird stuff would happen if cup_name and such were undefined for the reagent.

Cups use cup_name, cup_icon_state and so on. If none is given, they default to the aquivalent of the 'brown glass', where reagents cannot be identified with the eye.

**Comment by Zuhayr** [comment]
Tested this locally, anything other than the various coffee things defaults to 'cup of ...what?'.

The sprites are pretty... lacking, imo, but they will do for an initial commit.

**Pull request merged**